File: bilibili/src/main/java/com/bilibili/ui/recommed/RecommendContract.java
package com.bilibili.ui.recommed;

import com.common.base.BasePresenter;
import com.common.base.BaseView;

import me.drakeet.multitype.Items;

/**
 * Created by miserydx on 17/7/6.
 */

public interface RecommendContract {

    interface View extends BaseView {

        void onDataUpdated(Items items, boolean isLoadMore);

        void onRefreshingStateChanged(boolean isRefresh);

    }

    interface Presenter extends BasePresenter {

        void loadMore();

    }

}

File: bilibili/src/main/java/com/bilibili/ui/recommed/RecommendContract.kt
package com.bilibili.ui.recommed

import com.common.base.BasePresenter
import com.common.base.BaseView

import me.drakeet.multitype.Items

/**
 * Created by miserydx on 17/7/6.
 */

interface RecommendContract {

    interface View : BaseView {

        fun onDataUpdated(items: Items, state: Int)

        fun onRefreshingStateChanged(isRefresh: Boolean)

    }

    interface Presenter : BasePresenter {

        fun pullToRefresh(idx: Int)

        fun loadMore(idx: Int)

        fun loadingMoreFinished()

    }

}
